test/passing/gen/gen.ml

```diff
@@ -51,20 +59,24 @@ let emit_test test_name setup =
     match setup.base_file with Some n -> n | None -> test_name
   in
   let extra_deps = String.concat " " setup.extra_deps in
+  let cmd_prefix = if setup.should_fail then "! " else "" in
+  let cmd =
+    Printf.sprintf "%%{bin:ocamlformat}%s %%{dep:%s}" opts base_test_name
+  in
   Printf.printf
     {|
 (rule
  (targets %s.output)
  (deps .ocamlformat %s)
  (action
    (with-outputs-to %%{targets}
-     (system "%%{bin:ocamlformat}%s %%{dep:%s} || true"))))
+     (system "%s%s"))))
```

Comment:
Could switch to `run` when `should_fail` is not set but that would make the code slightly more complicated.

Reply:
I tried, but unfortunately the `%{read-lines}` expands to a single string with spaces in between, so ocamlformat gets a large string argument. We could solve that by not doing `%{read-lines}` and generating the actual options in `dune.inc`, but that makes a larger file.

Reply:
I see. I think that's OK, we can merge like this,